### DISCUSSION TITLE
BLE Scan Results - Match with previously found devices

### DISCUSSION THREAD

**Discussion by RobSchein** [shinyorg/shiny#1525] [Q&A] [answered]
Hi Folks,
Per the FAQs...UUIDs should be treated on a "per scan" basis.  My app might want to connect to two or more of the same device & I provide a way to rename those to user-supplied names.  I squirrel those away in a file locally.  When I reinvoke my app, I was planning on using the squirrelled away UUID/user names to correlate the newly scanned device to its user-friendly name from that file.  Doesn't sound like I should be doing that :)

Any suggestions on some correlation mechanism between my device and the user-supplied names that will survive invocations?

Thanks in advance!
-Rob

**Comment by aritchie** [comment, marked as answer]
This answer depends on many things but it outside of plugin.  If you have control of the peripheral, you can broadcast something like a mac address or ID on the manufacturer data.  Always advertise a custom service UUID that you can use to discover your BRAND of devices.  Whatever you do.... do not save the peripheral UUID anywhere in your code.  It will change.  That should be used as a temporary pointer and nothing more

**Reply by RobSchein** [reply]
Thanks, Allan.  Your suggestion of using the mac address is exactly where I had landed.  I was just wondering whether there was a standard that folks used.  As always, thanks for your input!
-Rob
